motorlib/motor.py

```diff
@@ -1,177 +1,289 @@
+            raise ValueError(
+                "Ideal pressure cannot be calculated. Propelant is missing."
+            )
+
         _, _, gamma, T, molarMass = self.propellant.getCombustionProperties(chamberPres)
 
-        if chamberPres <= atmosphericPressure: # Mach calculation gets weird at low chamber pressures
+        if (
+            chamberPres <= atmosphericPressure
+        ):  # Mach calculation gets weird at low chamber pressures
             return 0
 
-        def machFunc(M, chamberPres, massFlux, gamma, T, molarMass, gasConstant):
-            A = chamberPres * (gamma * molarMass / (gasConstant * T)) ** 0.5
-            B = 1.0 + ((gamma - 1.0) / 2.0) * M**2
-            C = -(gamma + 1.0) / (2.0 * (gamma - 1.0))
-            return A * M * (B ** C) - massFlux
-
-        def machFuncDerivative(M, chamberPres, massFlux, gamma, T, molarMass, gasConstant):
-            A = chamberPres * (gamma * molarMass / gasConstant / T) ** 0.5
-            B = 1.0 + ((gamma - 1.0) / 2.0) * M**2
-            C = -(gamma + 1.0) / (2.0 * (gamma - 1.0))
-            dB_dM = (gamma - 1.0) * M
-            return A * (B**C + M * C * (B**(C - 1.0)) * dB_dM)
-
-        maxMassFlux = machFunc(1.0, chamberPres, massFlux, gamma, T, molarMass, gasConstant) + massFlux
-
-        if massFlux >= maxMassFlux: # Boom
+        maxMassFlux = (
+            machFunc(
+                machNumber=1.0,
+                chamberPressure=chamberPres,
+                massFlowRation=massFlux,
+                gamma=gamma,
+                staticTemperature=T,
+                molarMass=molarMass,
+                gasConstant=gasConstant,
+            )
+            + massFlux
+        )
+
+        if massFlux >= maxMassFlux:  # Boom
             return 1.0
 
         x0 = np.arcsin(massFlux / maxMassFlux) * 2 / np.pi
-        M = newton(machFunc, fprime=machFuncDerivative, x0=x0, args=(chamberPres, massFlux, gamma, T, molarMass, gasConstant))
+        M = newton(
+            machFunc,
+            fprime=machFuncDerivative,
+            x0=x0,
+            args=(chamberPres, massFlux, gamma, T, molarMass, gasConstant),
+        )
 
         return max(M, 0)
 
-    def runSimulation(self, callback=None):
-        """Runs a simulation of the motor and returns a simRes instance with the results. Constraints are checked,
-        including the number of grains, if the motor has a propellant set, and if the grains have geometry errors. If
-        all of these tests are passed, the motor's operation is simulated by calculating Kn, using this value to get
-        pressure, and using pressure to determine thrust and other statistics. The next timestep is then prepared by
-        using the pressure to determine how the motor will regress in the given timestep at the current pressure.
+    def runSimulation(self, callback=None) -> SimulationResult:
+        """
+        Runs a simulation of the motor and returns a simRes instance with the results.
```

Comment:
I guess breaking these up ("google style"?) could make them easier to read, but I don't think they were all that long to begin with. Maybe revert this for now?